CoastalResilienceNetwork/GeositeFramework
# Add support for zooming to layer extent

Built on top of #537 

* Clicking on the ellipse icon (temporary location) for a layer
zooms to the extent of that layer.
* Add support to get and store details/metadata for a layer.
* Add convenience method to Layer that returns an Extent object
for a layer if the layer details have been retrieved.

**Testing instructions**
- Open the network console.
- Before activating a layer in the Coastal Resilience group, click the ellipse icon next to a layer. The map should zoom to the extent of the layer (around New Jersey). In the network tab, you should see one request to the map service, and one request for the layer details.
- Click the ellipse next to a different layer in the same group. You should see only one request for the layer details.
- Activate a layer in the group. There should be no requests for the service or the layer details, only the tiles and the legend.
- Zoom to the extent of one of the layers you zoomed to before. There should be no requests.

Connects to #520
